src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -222,6 +226,45 @@ trait EntityComponent {
       entityAttributeQuery.batchInsertAttributes(attributeRecsToEntityId.keys.toSeq)
     }
 
+    private def upsertAttributes(workspaceContext: SlickWorkspaceContext, entities: Traversable[Entity]) = {
+      DBIO.sequence(entities.map { entity =>
+        getWithId(workspaceContext, entity.entityType, entity.name) flatMap { case loadedEntity =>
```

Comment:
this whole thing is 1 query to delete all attributes that we want to delete (0 if there are none to delete), 1 query to insert all new attributes (0 of there are none to insert), and n queries where n is the number of batches supplied in the API call (this would generally scale linearly with the number of entities that we're updating, but users could be weird).

i believe i could get the n queries to be a constant 1 by feeding in the attribute records that we have already loaded, but the could would get even messier. thoughts?